rust/src/lib.rs

```diff
@@ -1306,30 +1340,48 @@ pub unsafe extern "C" fn zcashlc_derive_transparent_address_from_seed(
         } else {
             return Err(format_err!("index argument must be positive"));
         };
-        let sk = derive_secret_key_from_seed(&network, &seed, AccountId(account), index);
-        let taddr = derive_transparent_address_from_secret_key(&sk.unwrap()).encode(&network);
+        let tfvk = UnifiedSpendingKey::from_seed(&network, &seed, AccountId::from(account))
+            .map_err(|e| format_err!("error generating unified spending key from seed: {:?}", e))
+            .map(|usk| usk.transparent().to_account_pubkey())?;
+        let taddr = match p2pkh_addr(tfvk, index) {
+            Ok(taddr) => taddr,
+            Err(e) => return Err(format_err!("Couldn't derive transparent address: {:?}", e)),
+        };
+        let taddr = taddr.encode(&network);
 
         Ok(CString::new(taddr).unwrap().into_raw())
     });
     unwrap_exc_or_null(res)
 }
 
-/// Derives a transparent address from the given secret key enconded as a WIF string
+/// Derives a transparent address from the given account private key.
 #[no_mangle]
-pub unsafe extern "C" fn zcashlc_derive_transparent_address_from_secret_key(
-    tsk: *const c_char,
+pub unsafe extern "C" fn zcashlc_derive_transparent_address_from_account_private_key(
+    xprv: *const c_char,
```

Comment:
This is only used on `shield_funds` 
unless the wallet is storing this private key, we should refactor the SDK's public API to require  USK for shielding funds and this could be removed and replaced with migration documentation